src/Compilers/CSharp/Portable/Syntax/InternalSyntax/TypeSyntax.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax
+{
+    internal abstract partial class TypeSyntax
+    {
+        public bool IsVar
```

Comment:
There is a similar implementation on the public `TypeSyntax` (in `D:\repos\roslyn\src\Compilers\CSharp\Portable\Syntax\TypeSyntax.cs`). Maybe it could be factored to call this one or be removed completely?